File: lib/core/models/notification.dart
class Notification {
  final String idNotification;
  final String? idUser;
  final String title;
  final String type; // Cập nhật ứng tuyển, Phỏng vấn, Việc làm mới
  final DateTime dateTime;
  final String status; // Đã gửi, Lên lịch, Chờ xử lý
  final String actionUrl;
  final DateTime createdAt;

  Notification({
    required this.idNotification,
    this.idUser,
    required this.title,
    required this.type,
    required this.dateTime,
    required this.status,
    required this.actionUrl,
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      idNotification: json['idNotification'],
      idUser: json['idUser'],
      title: json['title'],
      type: json['type'],
      dateTime: DateTime.parse(json['dateTime']),
      status: json['status'],
      actionUrl: json['actionUrl'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idNotification': idNotification,
      'idUser': idUser,
      'title': title,
      'type': type,
      'dateTime': dateTime.toIso8601String(),
      'status': status,
      'actionUrl': actionUrl,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: HuitWorks.WebAPI/Models/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HuitWorks.WebAPI.Models
{
    [Table("notifications")]
    public class Notification
    {
        [Key]
        [Column("idNotification")]
        [StringLength(50)]
        public string IdNotification { get; set; } = null!;

        [Column("idUser")]
        [StringLength(64)]
        public string IdUser { get; set; } = null!;

        [Column("title")]
        [StringLength(255)]
        [Required(ErrorMessage = "Tiêu đề là bắt buộc.")]
        public string Title { get; set; } = null!;

        [Column("type", TypeName = "enum('Cập nhật ứng tuyển','Phỏng vấn','Việc làm mới')")]
        public string Type { get; set; } = null!;

        [Column("dateTime")]
        public DateTime DateTime { get; set; }

        [Column("status", TypeName = "enum('Đã gửi','Lên lịch','Chờ xử lý', 'Đã xử lý')")]
        public string Status { get; set; } = null!;

        [Column("actionUrl")]
        [StringLength(255)]
        public string? ActionUrl { get; set; }

        [Column("createdAt")]
        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("isRead")]
        [Required]
        public int IsRead { get; set; } = 0;
    }
}
